addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -521,6 +524,9 @@ export class LinkPlugin extends Plugin {
                 this.linkInDocument = null;
                 this.currentOverlay.close();
             },
+            onEdit: () => {
+                this.restoreSavePoint = this.dependencies.history.makeSavePoint();
+            },
```

Comment:
I didn't find a better way to do it without adding a new prop, so let's keep this.

By the way, since we also have the discard issue with 18.3, if it's not too much trouble, could you cherry-pick only this part of code to add an onEdit savepoint and make a branch targeting to 18.3? :face_with_peeking_eye: Remember that you will need to have a good commit message when working on stable branches.

I didn't find an existing task in our pipe so I will ping @dmo-odoo just in case.